### PR TITLE
Tests: rework `test_cap_kcub1`

### PR DESCRIPTION
**Pull request by dshil**
* Rework the `_check_cap_kcub1` method.
* Add a check for the end of a line.

### CODE REVIEW

File: gits/test/helper.py [hunk 1]
```diff
@@ -477,18 +477,22 @@ def _check_cap_il1(self, s, pos):
...
-        self.assertEqual(want_cur_x, self._terminal._cur_x)
-        self._terminal._cap_rs1()
+        term = self._terminal
+        x, _ = pos
+        term._cur_x, term._cur_y = pos
+
+        term._cap_kcub1()
+
+        self.assertFalse(term._eol)
+        self.assertEqual(x - 1, term._cur_x)
```
Comment (eugulixes) [inline gits/test/helper.py]: We have to be able to check 

`self._check_cap_kcub1((0, 0))`

and make sure that `x` doesn't change after calling `_cap_kcub1`.
Howeve, in this case we will get `-1`.

Comment (dshil) [reply gits/test/helper.py]: fxd, ping @eugulixes

File: gits/test/capabilities_test.py [hunk 2]
```diff
@@ -482,16 +482,18 @@ def test_cap_kcub1(self):
         """
         term = self._terminal
 
+        # The x position of the cursor is at the left-most corner.
```
Comment (eugulixes) [inline gits/test/capabilities_test.py]: Let's remove this comment. The code is very straightforward.
The following comment is wrong. Probably it's better to remove it as well.
The rest of the comments are ok. However, some of them should be removed in a different PR.

Comment (dshil) [reply gits/test/capabilities_test.py]: ping @eugulixes 
Agree. Do we need to have an issue for this enhancement?

Comment (eugulixes) [reply gits/test/capabilities_test.py]: @dshil I'll do the changes a bit later. Let's not create an issue for this.

**Comment by dshil** [comment]
ping @eugulixes, comment were taken into account,
I'm ready to another round of the review :)

File: gits/test/capabilities_test.py [hunk 3]
```diff
@@ -492,16 +492,17 @@ def test_cap_kcub1(self):
         """
         term = self._terminal
 
+        self._check_cap_kcub1((0, 0))
+
         # The x position of the cursor is at the left-most position.
```
Comment (eugulixes) [inline gits/test/capabilities_test.py]: The comment is incorrect, but it's not worth correcting it. Let's remove it at all.

Comment (dshil) [reply gits/test/capabilities_test.py]: @eugulixes, fxd

**Review by eugulixes — APPROVED** [review]
